src/server/server.h

```diff
@@ -380,7 +381,7 @@ class Server {
   std::atomic<uint64_t> total_clients_{0};
 
   // slave
-  std::mutex slave_threads_mu_;
+  std::shared_mutex slave_threads_mu_;
```

Comment:
Would you like to split changes related to `slave_threads_mu_` to a new PR (e.g. with a title like `optimize slave threads via rwlock`)?

It seems not so related to the `WAIT` command?

Reply:
These changes are quite straightforward, but many lines are involved so reviewers may be overwhelmed and may not able to concentrate on WAIT changes 🤔